proposals/xxxx-mesh-nodes.md

```diff
@@ -0,0 +1,329 @@
+| *none* | Input record can be omitted when there is no record content. `[NodeDispatchGrid(...)]` is then required to specify a fixed grid size. |
+
+One system value is supported inside the *recordType* for `SV_DispatchGrid`,
+see [here](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#sv_dispatchgrid)
+for more details, where mesh nodes are just like broadcast launch nodes in
+this regard.
+Any other semantics on fields in the record structure are ignored.
+
+See the work graphs spec
+[Node input declaration](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#node-input-declaration)
+section under options for broadcasting launch for more detail.
+
+##### Input System Values
+
+For input values outside the input record, mesh nodes support the `broadcasting` launch [node shader system values](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#node-shader-system-values).
+These are the same [system values supported by mesh shaders](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/MeshShader.md#hlsl-attributes-and-intrinsics).
+
+| system value semantic | type    | description |
+|-----------------------|---------|-------------|
+| `SV_GroupThreadID`    | `uint3` | Thread ID within group |
+| `SV_GroupIndex`       | `uint`  | Flattened thread index within group |
+| `SV_GroupID`          | `uint3` | Group ID within dispatch |
+| `SV_DispatchThreadID` | `uint3` | Thread ID within dispatch |
+| Not supported for now: `SV_ViewID` | `uint` | See [definition in Mesh Shader spec](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/MeshShader.md#sv_viewid) |
+
+##### Node Outputs
+
+No outputs to other nodes are allowed for `[LaunchMode("mesh")]`.
+
+##### Mesh Shader Outputs
+
+The mesh node entry function also supports the [shared output arrays](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/MeshShader.md#shared-output-arrays) from mesh shader.
+The same definitions and rules from the mesh shader spec apply here.
+These arrays address memory shared across the group of threads, just like
+`groupshared` memory.
+The intrinsic function [`SetMeshOutputCounts`](#setmeshoutputcounts) must be called before writing to
+any of the output arrays.
+
+| Mesh shader shared output arrays                      | Array dimension defines maximum number of | Required  |
+|-|-|-|
+| vertex [`indices`](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/MeshShader.md#vertex-indices)                   | primitives    | Required  |
+| attributes for [`vertices`](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/MeshShader.md#vertex-attributes)       | vertices      | Required  |
+| attributes for [`primitives`](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/MeshShader.md#primitive-attributes)  | primitives    | Optional  |
```

Comment:
another nit: unlike above where the "Required" column is filled in with Y and N, here we have the "Required" and "Optional". I don't really care which, but I prefer consistency.